lib/build.js

```diff
@@ -180,22 +180,53 @@ export default {
 
       this.child.on('close', (exitCode) => {
         this.child = null;
-        this.errorMatcher.set(target.errorMatch, cwd, stdout + stderr);
+        this.errorMatcher.set(target, cwd, stdout + stderr);
 
         let success = (0 === exitCode);
         if (atom.config.get('build.matchedErrorFailsBuild')) {
-          success = success && !this.errorMatcher.hasMatch();
+          success = success && !this.errorMatcher.getMatches().some(match => match.criticality === 'error' || match.type && match.type.toLowerCase() === 'error');
+        }
+
+        function extractRange(json) {
+          return [
+            [ (json.line || 1) - 1, (json.col || 1) - 1 ],
+            [ (json.line_end || json.line || 1) - 1, (json.col_end || json.col || 1) - 1 ]
+          ];
+        }
+        function normalizePath(p) {
+          return require('path').isAbsolute(p) ? p : require('path').join(cwd, p);
+        }
+        function typeToSeverity(type) {
+          switch (type && type.toLowerCase()) {
+            case 'error': return 'error';
+            case 'warning': return 'warning';
+            case 'trace': return 'info';
+            case 'note': return 'info';
+            default: return null;
```

Comment:
Hmm, I have concerns about this. where did `trace` and `note` come from ? It doesn't seem to have anything to do with the generic `build` tool. SInce linter supports `error`, `warning` and `info` I think we should keep those levels too.

But this comment would be unnecessary if we didn't have both `type` and `severity`.